src/util/cast.js

```diff
@@ -20,6 +20,17 @@ class Cast {
      * @return {number} The Scratch-casted number value.
      */
     static toNumber (value) {
+        // If value is already a number we don't need to coerce it with
+        // Number().
+        if (typeof value === 'number') {
+            // Scratch treats NaN as 0, when needed as a number.
+            // E.g., 0 + NaN -> 0.
+            if (isNaN(value)) {
```

Comment:
Is `Number.isNaN(value)` more efficient for a similar reason? `isNaN` still implicitly coerces to a number.

Reply:
@joker314, good catch!

Reply:
@joker314 That's a great point.

I made the change (with back compat support in mind for Internet Explorer which does not provide Number.isNaN, in which case self.isNaN is a safe backup) and recorded some new numbers.

|  Project, Platform | before (% of time) | after (% of time) | Change |
|  ------ | ------ | ------ | ------ |
|  89811578, 2017 MBP 15" Chrome | 1.50% | 0.93% | 0.57% |
|  89811578, 2017 MBP 15" Firefox | 1.79% | 1.24% | 0.55% |
|  89811578, 2017 MBP 15" Safari | 2.00% | 1.40% | 0.60% |
|  89811578, 2017 SMS Chromebook | 1.58% | 0.90% | 0.68% |
|  89811578, 2018 Raspberry Pi 3B+ | 1.33% | 0.95% | 0.38% |

Chrome sees a minor change with it.
Looks like I could have recorded my Firefox numbers better.
And Safari sees an improvement now.

Reply:
I figure I'd construct a table to show the change for just this commit. (I think I worked this out correctly?)

<table>
<tr><th>Project, Platform</th><th>isNaN (%)</th><th>Number.isNaN (%)</th><th>Change of Change (%)</tr>
<tr><td>89811578, 2017 MBP 15" Chrome</td><td>0.46</td><td>0.57</td><td><strong>+0.11</strong></td></tr>

<tr><td>89811578, 2017 MBP 15" Firefox</td><td>0.63</td><td>0.55</td><td><strong>-0.08</strong></td></tr>

<tr><td>89811578, 2017 MBP 15" Safari</td><td>0.00</td><td>0.60</td><td><strong>+0.60</strong></td></tr>

<tr><td>89811578, 2017 SMS Chromebook</td><td>0.68</td><td>0.68</td><td><strong>±0.00</strong></td></tr>

<tr><td>89811578, 2018 Raspberry Pi 3B+</td><td>0.21</td><td>0.38</td><td><strong>+0.17</strong></td></tr>
</table>